llarp/config/ini.cpp

```diff
@@ -168,4 +169,33 @@ namespace llarp
     m_Overrides.clear();
   }
 
+  void
+  ConfigParser::SaveNew()
+  {
+    if (not m_Overrides.empty())
+    {
+      throw std::invalid_argument("Override specified when attempting new .ini save");
+    }
+    if (m_Config.empty())
+    {
+      throw std::invalid_argument("New config not loaded when attempting new .ini save");
+    }
+    if (m_FileName.empty())
+    {
+      throw std::invalid_argument("New config cannot be saved with filepath specified");
+    }
+
+    std::ofstream ofs(m_FileName);
+    for (const auto& [section, values] : m_Config)
+    {
+      ofs << std::endl << "[" << section << "]" << std::endl;
+      for (const auto& [key, value] : values)
+      {
+        ofs << key << "=" << value << std::endl;
+      }
+    }
+
+    m_Config.clear();
```

Comment:
why do we want to call `clear()` here?

Reply:
though that's a good point... i imagine the other functions call clear() to make the object re-usable, but this configparser object is destroyed after the call to config endpoint is completed

Reply:
this also makes this able to be const